### DISCUSSION TITLE
Need help with MQ135 Gas sensor library

### DISCUSSION THREAD

**Discussion by mysteryman6174** [micropython/micropython#11431] [Libraries & Drivers]
Hello there, 

I've just stumbled upon the mq135 library at [MQ35Libarary by Rubfi](https://github.com/rubfi/MQ135)
Initially i was getting math domain error but upon looking it up i found a archived micro-python forum post at [Micro python Post](https://forum.micropython.org/viewtopic.php?t=12206&p=66233), i used his code and it seemed to work but upon closer inspection the PPM values are very weird, the sensor is connected to GPIO 34

I have no idea whats causing this, I've tried rechecking the pins and setup everything is fine but the values never change, the output is attached below
![image](https://user-images.githubusercontent.com/116084737/236588624-66401336-c285-4996-a955-690eadada12e.png)

**Comment by andrewjosephlehmann** [comment]
I am having the very same issue with my raspberry pi pico w.

**Reply by mysteryman6174** [reply]
It could be faulty hardware, although I'm not sure, try reading ADC pin manually to which the sensor is connected to

**Comment by andypiper** [comment]
What board and sensor are you using and how is it all connected up? What version of MicroPython are you on?

**Reply by mysteryman6174** [reply]
The board I'm using is the ESP-WROOM-32,
The sensor is connected to VIN pin of esp32, the esp is USB powered,
Then the Analog A0 Pin of MQ135 is connected to a voltage divider circuit of 1K and 2K Ohms (5v to 3.3v).
The voltage divider output is then connected to Pin32 of the ESP-32

As for the micro-python version it's the latest version from the website, freshly installed,
Further looking into it I've found that the ADC.read() value for it always returning 0, could it be a hardware fault ?
To test this I've done
mq = machine.ADC(machine.Pin(34))
mq.read() & mq.read_u16()
Both of them return 0

**Reply by andypiper** [reply]
Just to check, you have it connected to Pin 32 but the code refers to Pin 34… typo? (You did refer to GPIO 34 in the earlier post, so probably just a typo in this comment?)

**Reply by mysteryman6174** [reply]
Oh I'm sorry its actually pin 34 that I've connected the mq135 sensor to, the previous comment is the typo the code is correct

**Reply by andypiper** [reply]
I think I have an MQ135 somewhere in my studio so I’ll try to replicate this (if I can find it!). Bear with me.

**Reply by mysteryman6174** [reply]
That will be great, I'll wait for your results.
